test/pkg/environment/azure/environment.go

```diff
@@ -51,6 +51,7 @@ func (env *Environment) DefaultNodePool(nodeClass *v1alpha2.AKSNodeClass) *corev
 	nodePool.Spec.Template.Spec.NodeClassRef = &corev1beta1.NodeClassReference{
 		Name: nodeClass.Name,
 	}
+	nodePool.Spec.Weight = lo.ToPtr[int32](1)
```

Comment:
Is this needed to make it higher priority than the default NAP pool, or what exactly is this weight impacting?

Reply:
# Priority given to the NodePool when the scheduler considers which NodePool
  # to select. Higher weights indicate higher priority when comparing NodePools.
  # Specifying no weight is equivalent to specifying a weight of 0.
  weight: 10
  
  
 we want it to be higher weighted than the system surge and other pools since we are testing specific nodepool functionality. I could set this to 10 instead